ouroboros-consensus/test-consensus/Test/Dynamic/Praos.hs

```diff
@@ -88,8 +88,7 @@ prop_simple_praos_convergence params numCoreNodes numSlots =
     prop_simple_protocol_convergence
       (\nid -> protocolInfo numCoreNodes nid (DemoPraos params))
       isValid
-      numCoreNodes
-      numSlots
+      (immediateJoinConnectionPlan numCoreNodes numSlots)
```

Comment:
Unlike BFT eg, the existing Praos tests fix the number of nodes and the number of slots. Because they seemed so controlled, I conservatively chose not to introduce delays in them -- at least not without consulting you. What seems appropriate to you here?

On the technical side, I'd need to write a different `ConnectoinPlan` generator that respects a fixed number of nodes and/or fixed number of slots (probably using QuickCheck's `forAll`, unless the team is amenable to `KnownNat` contexts).

Reply:
Ok, we'll deal with Praos later.

`forAll` prevents shrinking, so we don't want that. We could introduce a type that combines the number of nodes/slots and the connection plan.

Reply:
Resolved by writing a `genConnectionPlan` that respects a given `NumCoreNodes` and `NumSlots` parameter.